sycl/source/detail/jit_compiler.cpp

```diff
@@ -694,6 +701,20 @@ jit_compiler::fuseKernels(QueueImplPtr Queue,
       return A.MIndex < B.MIndex;
     });
 
+    // Determine whether the kernel has been subject to DPCPP's range rounding.
+    // If so, the first argument will be the original ("user") range.
+    std::optional<size_t> UserGlobalSize;
+    if ((KernelName.find("_ZTSN4sycl3_V16detail18RoundedRangeKernel") == 0 ||
+         KernelName.find("_ZTSN4sycl3_V16detail19__pf_kernel_wrapper") == 0) &&
+        !Args.empty() &&
+        Args[0].MType == kernel_param_kind_t::kind_std_layout && Args[0].MPtr &&
+        Args[0].MSize == sizeof(size_t)) {
+      size_t UGS = *reinterpret_cast<size_t *>(Args[0].MPtr);
+      assert(KernelCG->MNDRDesc.Dims == 1 &&
```

Comment:
Does this mean that range rounding only happens for 1-dimensional kernels?

Reply:
Ah good catch, I misread that in the docs. No, it can happen on multi-dimensional kernels, however only the first dimension is rounded up. Fixed and tests added.